weedcoco/repo/deposit.py

```diff
@@ -0,0 +1,110 @@
+import argparse
+import pathlib
+import json
+import os
+import re
+from shutil import copyfile
+from weedcoco.utils import get_image_average_hash
+from weedcoco.validation import ValidationError, validate
+
+
+def validate_duplicate_images(image_hash):
+    if len(get_hashset_from_image_name(image_hash)) != len(image_hash):
+        raise ValidationError("There are identical images in the image directory.")
+
+
+def validate_existing_images(repository_dir, image_hash):
+    addition_hash = get_hashset_from_image_name(image_hash)
+    all_existing_hash = get_all_existing_hash(repository_dir)
+    if len(all_existing_hash.union(addition_hash)) != len(all_existing_hash) + len(
+        addition_hash
+    ):
+        raise ValidationError("There are identical images in the repository.")
+
+
+def get_hashset_from_image_name(image_hash):
+    return set(
+        map(lambda image_name: os.path.splitext(image_name)[0], image_hash.values())
+    )
```

Comment:
I think it is more pythonic, and more readable, to express this as
```suggestion
    return {os.path.splitext(image_name)[0] for image_name in image_hash.values()}
```